### ISSUE TITLE
Fix party for top known MP but unknown party

### ISSUE THREAD

**Issue by fredrik1984** [welfare-state-analytics/riksdagen-corpus#324]
Based on the 0.9 corpus version, we should fix the top known MP but with an unknown party. Most of these MPs are from the 19th century. The top 100 known MPs but with an unknown party account for 22 000 hits. Adding party to each MP's Wikidata post would perhaps be a start. 

@salgo60 is this something that you would like to take on?

[known_person_but_unknown_party.csv](https://github.com/welfare-state-analytics/riksdagen-corpus/files/11829359/known_person_but_unknown_party.csv)

**Comment by salgo60** [comment]
@fredrik1984 My focus is to connect people to the book Två kammar Riksdagen right now about 600 to do see https://github.com/salgo60/Wikidata_riksdagen-corpus/issues/38#issuecomment-1360364007

* [SPARQL saknar "Tvåkammar-riksdagen 1867-1970" as a Source](https://w.wiki/6p78) = 602

I can try to follow your ranking list when I am back home this week

----
### unknown party
I use "no value" for people in the book that has no party set 

I am in a hammock out kayaking but try some SpARQL on my mobile 
* [PM with no party and not "no value"](https://w.wiki/6rrf) = 276
* [PM with party "no value"](https://w.wiki/6sLk) = 344

**Comment by salgo60** [comment]
Your 1st on the list https://www.wikidata.org/wiki/Q5616733

Was updated by me feb 2022 and then I didn't add party from the book at SPA so there is a lot of cleaning needed ...


SPA for Q5616733 https://portrattarkiv.se/details/sj9PGLAlnmUAAAAAABgdHQ

----
We also have a lot of parties set by other Wikidata users that is not following "the book" I.e you can [filter on the timeline](https://js.histropedia.com/apps/query-timeline/index.html?q=SELECT%20distinct%20?person%20?personLabel%20(SAMPLE(?SPA)%20AS%20?SPA)%20%20(sample(?bild)%20AS%20?bild)%20%0A?birth%20?death%20?partyLabel%20WHERE%20%7B%0A%0A%20%20VALUES%20?member%20%7B%0A%20%20%20%20wd:Q33071890%20%0A%20%20%20%20wd:Q81531912%20%0A%20%20%20%20%23wd:Q82697153%20%0A%20%20%20%20wd:Q10655178%20%0A%20%20%7D%0A%20%20?person%20wdt:P39%20?member.%0A%20%20OPTIONAL%7B?person%20wdt:P102%20?party%7D%0A%20%20OPTIONAL%7B?person%20wdt:P569%20?birth%7D%0A%20%20OPTIONAL%7B?person%20wdt:P570%20?death%7D%0A%0A%0A%20%20OPTIONAL%20%7B?person%20wdt:P18%20?bild%7D.%0A%20%0A%20%20BIND(URI(CONCAT(%22https://portrattarkiv.se/details/%22,?SPAid))%20AS%20?SPA)%0A%0A%20%20SERVICE%20wikibase:label%20%7B%20bd:serviceParam%20wikibase:language%20%22sv,en%22.%20%7D%0A%7D%20GROUP%20BY%20%20?person%20?personLabel%20%20?death%20?birth%20?partyLabel%0Aorder%20by%20?partyLabel&md=true&g=article&l=person&t=personLabel&s=birth&e=death&i=bild&d=0&c=partyLabel&f=partyLabel&v=t) and see parties at the wrong Time period is my feeling

![image](https://github.com/welfare-state-analytics/riksdagen-corpus/assets/14206509/129b8aa1-52c4-498f-a604-c0f4ca6eec31)

**Comment by fredrik1984** [comment]
Yes, adding party is of course a bit tricky when it comes to the 19th century. And as we have discussed previously, we use the bio books as bible. The abbreviation register at the end of the bio books, and the hierarchical tree of party formations/splits, are good sources for this work. 

So in the case of Anders Danielsson (https://www.wikidata.org/wiki/Q5616733) he was a member of Lantmannapartiet (lmp) 1873–1887, Nya lantmannapartiet (nya lmp) 1884–1894), and then to the reappearing (!) Lantmannapartiet (lmp) 1895–1897.

**Comment by salgo60** [comment]
Yes and in the Anders case I "connected" him to the book and didn't add party

Now when I connect to the book I
* add party
* add SPA picture of the book
* corrects period - before I didn't change if we had a diff what was added before in WD
* add name in Riksdagens
* add incomes and lived in if possible..
*...

I have 11 m/s wind right now so we never know how my kayaking will end maybe we get another "Titanic submarine" news story 😅

**Comment by fredrik1984** [comment]
Ok! Good luck with the kayaking!

**Comment by salgo60** [comment]
## Unknow parties -  we need sources.....

@fredrik1984 the list you somehow has created - _I guess people identified speaking in the PM_
* People in this list not in Emils [TvåKammarBookList and EnkammarBookList](https://public-paws.wmcloud.org/27568293/MissingInBooks.csv) I have found --> we can have problems finding a good source....
   * e.g. 
0,[Q5622212](https://www.wikidata.org/wiki/Q5622212),389.0,,,,,,,left_only,,,,,left_only --> [Ulf Dinkelspiel](https://sv.wikipedia.org/wiki/Ulf_Dinkelspiel) - _är moderat gissar jag_ saknades i WD
1,[Q5991125](https://www.wikidata.org/wiki/Q5991125),255.0],,,,,,,left_only,,,,,left_only --> [Rune Molin](https://sv.wikipedia.org/wiki/Rune_Molin) - [gissar sosse](url)
2,[Q6077653](https://www.wikidata.org/wiki/Q6077653),253.0,,,,,,,left_only,,,,,left_only --> [Sven Romanus](https://sv.wikipedia.org/wiki/Sven_Romanus) - [partipolitiskt obunden](https://sv.wikipedia.org/wiki/Partipolitiskt_obunden)
3,[Q4961261](https://www.wikidata.org/wiki/Q4961261),222.0,,,,,,,left_only,,,,,left_only --> [Reidunn Laurén](https://sv.wikipedia.org/wiki/Reidunn_Laur%C3%A9n) - _opolitisk_
.......

#### WD modulering - opolitisk - partipolistisk obunden
Jag vet inte om WD har en bra modell för hur "opolitisk...." skall hanteras se [Reidunn Laurén](https://sv.wikipedia.org/wiki/Reidunn_Laur%C3%A9n).... 
* Spontant tror jag att man skall ha WD objekt för detta precis som jag [testat med "vilde"](https://github.com/salgo60/Wikidata_riksdagen-corpus/issues/139)
* Alla dessa varianter bör tas om hand i era tankar om partier...
   * tror en version 0.1 av detta dokument vore bra att få ut så vi har dokumenterat vad vi testar....

**Comment by salgo60** [comment]
**Status**: Touch and go

I have been home for some days and the following has been done,.... now I am leaving again...
1) @dpriskorn have given some care to [EntityShape](https://github.com/dpriskorn/entityshape) that I use in the below Notebook... [video](https://youtu.be/1orqJ3HkKF8)
1) we have since earlier played with ShEx see [#129](https://github.com/salgo60/Wikidata_riksdagen-corpus/issues/129)
   1) WIkimedia has a hosted Notebook environment [PAWS](https://hub-paws.wmcloud.org) --> I created a  Notebook "[Test validate people in Wikidata with a ShEx schema E395 from a list](https://public-paws.wmcloud.org/User:Salgo60/RiksdagensCorpusSchema_List.ipynb)" [video](https://youtu.be/1orqJ3HkKF8)
        1) that use your csv file mentioned above
        1) runs through the list and try to validate them with [E395](https://www.wikidata.org/wiki/EntitySchema:E395)
        1) As a lot of people you have in your list didnt appear in the books --> [we have a source problem](https://github.com/welfare-state-analytics/riksdagen-corpus/issues/324#issuecomment-1612672745) I also tried to compare your list with the book see [MissingInBooks.csv](https://public-paws.wmcloud.org/27568293/MissingInBooks.csv)
   
   <img width="1018" alt="image" src="https://github.com/welfare-state-analytics/riksdagen-corpus/assets/14206509/64089872-528e-4bd4-b27f-92b5bd4449c5">   
   
   <img width="1018" alt="image" src="https://github.com/welfare-state-analytics/riksdagen-corpus/assets/14206509/27648a81-f501-4494-a468-b7ffb1d521e1">

**Comment by salgo60** [comment]
### Konstigheter

1) **lmp t** wd [Q6009646](https://www.wikidata.org/wiki/Q6009646) [SPA](https://portrattarkiv.se/details/sj9PGLAlnmUAAAAAABfpQA)

<img width="1144" alt="image" src="https://github.com/welfare-state-analytics/riksdagen-corpus/assets/14206509/67399d32-46e4-4ef4-8a87-8c26d2b8f573">

**Comment by salgo60** [comment]
I am away for another longer kayak trip, guess back in 2-3 weeks.... **assign this to me salgo60 if you need support**

I feel we should
* [ ] formalize what data your project request from Wikidata in a [ShEx](https://shex.io/shex-primer/#tripleConstraint)
* [ ] formalize what trusted sources should be used to confirm the meta data
    * see queries below what sources used today in Wikidata for Swedish PM in P1343
    
### Wikidata sources on Swedish PM people using property "described by source" [Property:P1343](https://www.wikidata.org/wiki/Property:P1343) 
* SPARQL 
   * [table](https://w.wiki/6uCd)
   * [bubblechart](https://w.wiki/6uBx)
   * [barchart](https://w.wiki/6uCf)
   * [histropedia](https://js.histropedia.com/apps/query-timeline/index.html?q=SELECT%20distinct%20?person%20?personLabel%20(SAMPLE(?SPA)%20AS%20?SPA)%20%20(sample(?bild)%20AS%20?bild)%20%0A?birth%20?death%20?sourceLabel%20WHERE%20%7B%0A%0A%20%20VALUES%20?member%20%7B%0A%20%20%20%20wd:Q33071890%20%0A%20%20%20%20wd:Q81531912%20%0A%20%20%20%20wd:Q82697153%20%0A%20%20%20%20wd:Q10655178%20%0A%20%20%7D%0A%20%20?person%20wdt:P39%20?member;%0A%20%20%20%20wdt:P1343%20?source.%0A%20%20OPTIONAL%7B?person%20wdt:P102%20?party%7D%0A%20%20OPTIONAL%7B?person%20wdt:P569%20?birth%7D%0A%20%20OPTIONAL%7B?person%20wdt:P570%20?death%7D%0A%20%0A%0A%20%20OPTIONAL%20%7B?person%20wdt:P18%20?bild%7D.%0A%20%0A%20%0A%0A%20%20SERVICE%20wikibase:label%20%7B%20bd:serviceParam%20wikibase:language%20%22sv,en%22.%20%7D%0A%7D%20GROUP%20BY%20%20?person%20?personLabel%20%20?death%20?birth%20?sourceLabel%0Aorder%20by%20?sourceLabel&md=true&g=article&l=person&t=personLabel&s=birth&e=death&i=bild&d=0&c=sourceLabel&f=sourceLabel&v=t)
   
   [<img width="1190" alt="image" src="https://user-images.githubusercontent.com/14206509/221404045-fd7021b8-d1ca-4ca8-960c-5c6935021a7c.png">](https://js.histropedia.com/apps/query-timeline/index.html?q=SELECT%20distinct%20?person%20?personLabel%20(SAMPLE(?SPA)%20AS%20?SPA)%20%20(sample(?bild)%20AS%20?bild)%20%0A?birth%20?death%20?sourceLabel%20WHERE%20%7B%0A%0A%20%20VALUES%20?member%20%7B%0A%20%20%20%20wd:Q33071890%20%0A%20%20%20%20wd:Q81531912%20%0A%20%20%20%20wd:Q82697153%20%0A%20%20%20%20wd:Q10655178%20%0A%20%20%7D%0A%20%20?person%20wdt:P39%20?member;%0A%20%20%20%20wdt:P1343%20?source.%0A%20%20OPTIONAL%7B?person%20wdt:P102%20?party%7D%0A%20%20OPTIONAL%7B?person%20wdt:P569%20?birth%7D%0A%20%20OPTIONAL%7B?person%20wdt:P570%20?death%7D%0A%20%0A%0A%20%20OPTIONAL%20%7B?person%20wdt:P18%20?bild%7D.%0A%20%0A%20%0A%0A%20%20SERVICE%20wikibase:label%20%7B%20bd:serviceParam%20wikibase:language%20%22sv,en%22.%20%7D%0A%7D%20GROUP%20BY%20%20?person%20?personLabel%20%20?death%20?birth%20?sourceLabel%0Aorder%20by%20?sourceLabel&md=true&g=article&l=person&t=personLabel&s=birth&e=death&i=bild&d=0&c=sourceLabel&f=sourceLabel&v=t)
   
   
[<img width="729" alt="image" src="https://github.com/welfare-state-analytics/riksdagen-corpus/assets/14206509/e21520c5-1a2e-4cb9-8840-7ab8a6fd124e">](https://w.wiki/6uBx)

[<img width="1391" alt="image" src="https://github.com/welfare-state-analytics/riksdagen-corpus/assets/14206509/5314b5d4-e042-4fd2-8552-aa747c6c6b2e">](https://w.wiki/6uC3)

<img width="1391" alt="image" src="https://github.com/welfare-state-analytics/riksdagen-corpus/assets/14206509/d6a60405-bccc-4a20-ae40-a5e901067e26">

<img width="1391" alt="image" src="https://github.com/welfare-state-analytics/riksdagen-corpus/assets/14206509/50139266-54af-499c-9203-d13633df0617">

**Comment by fredrik1984** [comment]
Yes, we have to decide what to call these MPs that are "partipolitiskt obunden", "opolitisk", etc. We will consult the Riksdag Library about this after the summer. 

Thanks for helping out by adding parties to the MPs in the above list, this helps us a lot! And good luck on your kayak trip!

**Comment by salgo60** [comment]
> Yes, we have to decide what to call these MPs that are "partipolitiskt obunden", "opolitisk", etc. We will consult the Riksdag Library about this after the summer.
> 

@fredrik1984 the problem I see right now is that we are waiting on something and we see odd things in the data as I mentioned above - we need to agree on an approach **how to mark the data as questionable or need more care.**.. --> when we get the way forward it will be **easier to find those oddities and correct them**

### **Focus list Välfärden analyserad - parti created**
To get something started I mark those WD records were I see a potential problem with [P5008 on focus list of Wikimedia project](https://www.wikidata.org/wiki/Property:P5008) = [Q120143028 - Välfärden analyserad - parti](https://www.wikidata.org/wiki/Q120143028)

* [SPARQL list](https://w.wiki/6vqJ) - > 200 items right now

<img width="1378" alt="image" src="https://github.com/welfare-state-analytics/riksdagen-corpus/assets/14206509/d54379c7-e6eb-4ff3-b485-c75d33077597"><img width="1378" alt="image" src="https://github.com/welfare-state-analytics/riksdagen-corpus/assets/14206509/52fe8289-5d3e-4775-ad04-a3c93e03d6c8">

### Feels like a good pattern
A pattern like this feels good to "mark" objects that are cared about in different research projects...

**Comment by fredrik1984** [comment]
Hope you have had a nice summer with a lot of kayaking @salgo60! Just a question regarding this issue of adding party to the top known MPs but with unknown party: were you willing to add party to these MP wikidata posts?

**Comment by salgo60** [comment]
@fredrik1984 this is the list with potential problems... plus all the records regarding communists also feels need some more care...

### **Focus list Välfärden analyserad - parti created**
To get something started I mark those WD records were I see a potential problem with party [P5008 on focus list of Wikimedia project](https://www.wikidata.org/wiki/Property:P5008) = [Q120143028 - Välfärden analyserad - parti](https://www.wikidata.org/wiki/Q120143028)

* [SPARQL list](https://w.wiki/6vqJ) - > 200 items right now

<img width="1088" alt="image" src="https://github.com/welfare-state-analytics/riksdagen-corpus/assets/14206509/9eea40e0-2813-45a5-bdea-68ed5a3ee226">

**Comment by fredrik1984** [comment]
Thanks, I will try to take a look at this soon!

**Comment by BobBorges** [comment]
Closing as duplicate of #349 and #359
